### PR TITLE
[go-gen] Fix validation annotation

### PR DESCRIPTION
**Pull request by willjvsmith**
Annotations expected primitives, when they were checking pointers

### CODE REVIEW

**Review by c-harding — APPROVED** [review]

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala [hunk 1]
```diff
@@ -23,27 +23,27 @@ object GoServiceMainStructGenerator {
   private def generateValidatorAnnotation(attrType: AttributeType): String =
     s"`valid:${doubleQuote(
       attrType match {
-        case AttributeType.ForeignKey(_) | AttributeType.UUIDType => "type(string),uuid,required"
-        case AttributeType.BoolType                               => "type(bool),required"
-        case AttributeType.DateType                               => "type(string),required"
-        case AttributeType.DateTimeType                           => "type(string),rfc3339,required"
-        case AttributeType.TimeType                               => "type(string),required"
-        case AttributeType.BlobType(_)                            => "type(string),base64,required"
+        case AttributeType.ForeignKey(_) | AttributeType.UUIDType => "type(*string),uuid,required"
```
Comment (willjvsmith) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala]:
```suggestion
        case AttributeType.ForeignKey(_) | AttributeType.UUIDType => "type(*uuid.UUID),uuid,required"
```

Should this be this? Hard to know from docs

**Review by Lewiky — APPROVED** [review]
*LGTM

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala]: I'm pretty sure it should

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala [hunk 2]
```diff
@@ -23,27 +23,27 @@ object GoServiceMainStructGenerator {
   private def generateValidatorAnnotation(attrType: AttributeType): String =
     s"`valid:${doubleQuote(
       attrType match {
-        case AttributeType.ForeignKey(_) | AttributeType.UUIDType => "type(string),uuid,required"
-        case AttributeType.BoolType                               => "type(bool),required"
-        case AttributeType.DateType                               => "type(string),required"
-        case AttributeType.DateTimeType                           => "type(string),rfc3339,required"
-        case AttributeType.TimeType                               => "type(string),required"
-        case AttributeType.BlobType(_)                            => "type(string),base64,required"
+        case AttributeType.ForeignKey(_) | AttributeType.UUIDType => "type(*uuid.UUID),uuid,required"
```
Comment (jaylees14) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala]: Are you sure this works?

**Review by jaylees14 — APPROVED** [review]
Looks good, not sure about the UUID one though

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala (hunk 2 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala]: Nope, but I need Charlie's changes to test

Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainStructGenerator.scala]: The docs are shite

**Pull request merged**